crates/proto/src/op/lower_query.rs

```diff
@@ -13,7 +13,11 @@ use crate::rr::LowerName;
 use crate::rr::{DNSClass, RecordType};
 use crate::serialize::binary::*;
 
-/// Identical to [crate::op::Query], except that the Name is guaranteed to be in lower case form
+/// Identical to [Query], except that the Name is guaranteed to be in lower case form.
+///
+/// This guarantee is helpful for performance reasons, as case-sensitive comparisons
+/// can be done faster. Name comparison is a frequent operation and so benefits in sum
```

Comment:
I think this needs to be rephrased (same in other file)
```suggestion
/// can be done faster. Name comparison is a frequent operation and so overall performance benefits
```